src/meta/src/hummock/manager/mod.rs

```diff
@@ -659,7 +707,12 @@ impl HummockManager {
     ) -> Result<HummockSnapshot> {
         let snapshot = self.latest_snapshot.load();
         let mut guard = write_lock!(self, versioning).await;
-        let mut pinned_snapshots = BTreeMapTransaction::new(&mut guard.pinned_snapshots);
+        self.check_context_with_meta_node(context_id).await?;
```

Comment:
Question: at the very beginning version of code, we atomically check the context id when we commit to meta store. Now we only check the context id before we do the transaction. Will it cause bug if the context id becomes invalid after we have checked it but before we commit it to meta store?

Reply:
As mentioned by the comments in `check_context_with_meta_node`, "Caller should hold `versioning` lock, to sync with `HummockManager::release_contexts`". By adhering to that rule, correctness is guaranteed, because all callers interested in `context_id` are executed sequentially.

It's OK `context_id` has become invalid when commit to meta store.